src/lib/gui/widgets/FingerprintPreview.cpp

```diff
@@ -24,13 +23,18 @@ FingerprintPreview::FingerprintPreview(QWidget *parent, const QList<deskflow::Fi
   QString sha256Art;
 
   for (const auto &fingerprint : fingerprints) {
-    if (fingerprint.algorithm == "sha1") {
-      sha1String = QString::fromStdString(deskflow::formatSSLFingerprint(fingerprint.data));
-    }
-
-    if (fingerprint.algorithm == "sha256") {
-      sha256String = QString::fromStdString(deskflow::formatSSLFingerprintColumns(fingerprint.data));
-      sha256Art = QString::fromStdString(deskflow::generateFingerprintArt(fingerprint.data));
+    switch (fingerprint.type) {
+    case Fingerprint::Type::SHA1:
+      sha1String = deskflow::formatSSLFingerprint(fingerprint.data);
+      break;
+
+    case Fingerprint::Type::SHA256:
+      sha256String = deskflow::formatSSLFingerprintColumns(fingerprint.data);
+      sha256Art = deskflow::generateFingerprintArt(fingerprint.data);
+      break;
+    default:
+      qWarning() << "FingerprintPreview: Unknown fingerprint type: " << fingerprint.type;
```

Comment:
Log lines are always lower case in our app by convention. Also, we should use logging categories in future instead of hard coding class names in log strings (which usually don't get updated in future when classes are renamed).

```suggestion
      qWarning() << "unknown fingerprint type: " << fingerprint.type;
```